tmt/steps/report/reportportal.py

```diff
@@ -0,0 +1,62 @@
+import io
+import os
+import zipfile
+
+import junit_xml
+import requests
+
+import tmt.steps.report
+import tmt.utils
+
+from . import junit
+
+
+class ReportPortal(tmt.steps.report.ReportPlugin):
+    """
+    Send results in JUnit format to ReportPortal instance.
+
+    Requires environment variables TMT_REPORT_PORTAL_URL and TMT_REPORT_PORTAL_TOKEN.
+    """
+
+    _methods = [tmt.steps.Method(name="reportportal", doc=__doc__, order=60)]
+
+    def go(self) -> None:
+        """
+        Read executed tests, prepare junit, compress it to a zip zile and
+        send it to ReportPortal instance.
+        """
+        for env_var in ("TMT_REPORT_PORTAL_URL", "TMT_REPORT_PORTAL_TOKEN"):
+            if env_var not in os.environ:
+                raise tmt.utils.ReportError("Environment variable \"" + env_var + "\" not set")
+
+        super().go()
+
+        suite = junit.make_junit_xml(self)
+        data = junit_xml.TestSuite.to_xml_string([suite])
+        bytestream = io.BytesIO()
+
+        with zipfile.ZipFile(bytestream, "w", compression=zipfile.ZIP_DEFLATED,
+                             compresslevel=1) as zipstream:
+            # XML file names are irrelevant to Report Portal
+            with zipstream.open("tests.xml", "w") as entry:
+                entry.write(data.encode("utf-8"))
+        bytestream.seek(0)
+
+        response = requests.post(
```

Comment:
I would recommend performing `post()` under `tmt.utils.retry_session()`. One attempt or bust is not a good survival quality.